### PR TITLE
Ready for review

### PR DESCRIPTION
**Pull request by khomiakmaxim**
Telegram bot, which works similarly to [Splitwise app](https://www.splitwise.com/). Create groups, add members, so that all of them can track their expenses and see overal debt state in the group.

- Register bot with telegram [BotFather](https://t.me/BotFather), 
- Retrieve bot token
- Place it while running `cargo r -- --token "<YOUR TELEGRAM BOT TOKEN>"`

Written with [Rust](https://www.rust-lang.org/), [Teloxide](https://github.com/teloxide/teloxide) and [SeaORM](https://www.sea-ql.org/SeaORM/).

image
image

### CODE REVIEW

File: src/bot.rs [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+            if !expenses_in_group.is_empty() {
+                let mut user_spent: HashMap<String, Decimal> = HashMap::new();
+
+                for exp in expenses_in_group.iter() {
+                    *user_spent.entry(exp.username.clone()).or_default() += exp.amount;
+                }
+
+                // How much many was spent overall
+                let sum: Decimal = user_spent.iter().map(|x| *x.1).sum();
+                assert!(!user_spent.is_empty());
```
Comment (andrewalbrecht05) [inline src/bot.rs:302]: this line can cause a program to panic, it seems you wanted to do debug_assert

**Review by andrewalbrecht05 — APPROVED** [review]
This project looks good to me :+1: 
The code is well structured and easy to understand.
Also, i like the idea of this bot, seems quite useful

**Review by trepachkoDm — APPROVED** [review]
The "rust_splittea_bot" project shows a well-structured and comprehensive approach to building a bot with features for managing expenses in groups.

I liked its modular design, full CLI customization, and database integration.
Great job!

**Review by vsmysle — APPROVED** [review]
Nice job on the code! I have a couple of comments:

- It would be great to have tests covering the database interactions.
- The logging could be improved, especially with adding `tracing` calls.
- The commands appear to be lengthy and somewhat challenging to type (e.g `/addmembertgroup` or `/listexpensesingroup`). It might be beneficial to explore refactoring them with subcommands (if it is possible with the `teloxide` and Telegram).

**Review by chb19 — APPROVED** [review]

**Comment by khomiakmaxim** [comment]
> Nice job on the code! I have a couple of comments:
> 
> * It would be great to have tests covering the database interactions.
> * The logging could be improved, especially with adding `tracing` calls.
> * The commands appear to be lengthy and somewhat challenging to type (e.g `/addmembertgroup` or `/listexpensesingroup`). It might be beneficial to explore refactoring them with subcommands (if it is possible with the `teloxide` and Telegram).

Thanks, that indeed would be a good thing to do.

**Comment by khomiakmaxim** [comment]
> The "rust_splittea_bot" project shows a well-structured and comprehensive approach to building a bot with features for managing expenses in groups.
> 
> I liked its modular design, full CLI customization, and database integration. Great job!

Thanks, Dima!!

**Pull request merged**